### PR TITLE
Group entities in the navigation Bar

### PR DESCRIPTION
**Pull request by dmaciel**
Hi

The idea is to allow to group multiple entities into one option of the menu, this is helpful when there are too many entities or want to group based on relation. There is a example of the result:

navbargroup example

### CODE REVIEW

File: src/CRUDlex/ServiceProvider.php [hunk 1]
```diff
@@ -306,6 +307,24 @@ public function getEntities() {
...
+     */
+    public function getEntitiesNavBar() {
+      foreach ($this->datas as $entity => $data) {
+        $navBarGroup = $data->getDefinition()->getNavBarGroup();
+        if ($navBarGroup !== 'main'){
+          $result[$navBarGroup][] = $entity;
+        }else{
+          $result[$entity] = 'main';
+        }
+      }
```
Comment (philiplb) [inline src/CRUDlex/ServiceProvider.php]: Please indent with 4 spaces.

File: src/CRUDlex/ServiceProvider.php [hunk 2]
```diff
@@ -306,6 +307,24 @@ public function getEntities() {
     }
 
     /**
+     * Getter for the entitis fot the navigation bar.
+     *
+     * @return string[]
+     * a list of all available entity names with theirs group
```
Comment (philiplb) [inline src/CRUDlex/ServiceProvider.php]: "their"

File: src/CRUDlex/ServiceProvider.php [hunk 3]
```diff
@@ -306,6 +307,24 @@ public function getEntities() {
...
+     */
+    public function getEntitiesNavBar() {
```
Comment (philiplb) [inline src/CRUDlex/ServiceProvider.php]: This method should get an explicit test: EntityDefinitionTest::testGetEntitiesNavBar

File: src/CRUDlex/EntityDefinition.php [hunk 4]
```diff
@@ -557,6 +563,26 @@ public function setInitialSortAscending($initialSortAscending) {
...
         return $this->initialSortAscending;
     }
+    
+    /**
+     * Gets the navigation bar group where the entity belongs
+     *
+     * @return string
+     * the navigation bar group where the entity belongs
+     */
+    public function getNavBarGroup() {
```
Comment (philiplb) [inline src/CRUDlex/EntityDefinition.php]: This method should get an explicit test: EntityDefinitionTest::testGetNavBarGroup

File: src/CRUDlex/EntityDefinition.php [hunk 5]
```diff
@@ -557,6 +563,26 @@ public function setInitialSortAscending($initialSortAscending) {
...
+        return $this->navBarGroup;
+    }
+
+    /**
+     * Sets the navigation bar group where the entity belongs
+     *
+     * @param string $navBarGroup
+     * the navigation bar group where the entity belongs
+     */
+    public function setNavBarGroup($navBarGroup) {
```
Comment (philiplb) [inline src/CRUDlex/EntityDefinition.php]: This method should get an explicit test: EntityDefinitionTest::testSetNavBarGroup

File: docs/manual/extendedfeatures.rst [hunk 6]
```diff
@@ -140,6 +138,40 @@ Currently, the listview contains all entries on the pages. Often, it is desirabl
...
+too many entities or want to group based on relations between the entities, you
+can add a “navBarGroup” option. In this case we group library and books under
+“Learning Resources”:
+
+.. code-block:: yaml
+
+    library:
+        label: Library
+        table: library
+        navBarGroup: Learning Resources
```
Comment (philiplb) [inline docs/manual/extendedfeatures.rst]: The navBarGroup should be stated (and so validated) within the https://github.com/philiplb/CRUDlex/blob/master/src/definitionSchema.yml

**Comment by philiplb** [comment]
Hi,
thank you for the adjustments. :)
One of the tests is failing though: https://travis-ci.org/philiplb/CRUDlex/jobs/223153160

File: src/CRUDlex/EntityDefinition.php [hunk 7]
```diff
@@ -557,6 +563,26 @@ public function setInitialSortAscending($initialSortAscending) {
     public function isInitialSortAscending() {
         return $this->initialSortAscending;
     }
+    
+    /**
+     * Gets the navigation bar group where the entity belongs
```
Comment (philiplb) [inline src/CRUDlex/EntityDefinition.php]: Missing a dot at the sentence end.

File: src/CRUDlex/EntityDefinition.php [hunk 8]
```diff
@@ -557,6 +563,26 @@ public function setInitialSortAscending($initialSortAscending) {
...
+     *
+     * @return string
+     * the navigation bar group where the entity belongs
+     */
+    public function getNavBarGroup() {
+        return $this->navBarGroup;
+    }
+
+    /**
+     * Sets the navigation bar group where the entity belongs
```
Comment (philiplb) [inline src/CRUDlex/EntityDefinition.php]: Missing a dot at the sentence end.

File: src/CRUDlex/ServiceProvider.php [hunk 9]
```diff
@@ -306,6 +307,24 @@ public function getEntities() {
     }
 
     /**
+     * Getter for the entities fot the navigation bar.
```
Comment (philiplb) [inline src/CRUDlex/ServiceProvider.php]: "for"

File: docs/manual/extendedfeatures.rst (hunk 6 above)
Comment (philiplb) [inline docs/manual/extendedfeatures.rst]: This should get an entry within https://github.com/philiplb/CRUDlex/blob/master/docs/manual/crudyamlreference.rst

**Comment by dmaciel** [comment]
Hi

I already made the requested changes, but looks like this pull request is staled. Can you confirm if everything is ok?

**Comment by philiplb** [comment]
All good, merged now. :)
Thank you very much for this nice feature! :)

**Pull request merged**